src/allocator.rs

```diff
@@ -15,9 +15,12 @@ pub struct AtomBuf {
 }
 
 impl AtomBuf {
-    pub fn is_empty(self) -> bool {
+    pub fn is_empty(&self) -> bool {
         self.start == self.end
     }
+    pub fn len(&self) -> usize {
```

Comment:
You might consider exposing this on `LazyNode` with `.atom_length` which returns `int` for an atom and `None` for a pair.

Reply:
yes, I have been considering that. I will probably propose some patches to take this short cut internally first, to not go through a `&[u8]` just to get the length.